Point_set_processing_3/include/CGAL/structure_point_set.h

```diff
@@ -0,0 +1,1483 @@
+  std::vector<Edge> m_edges;
+  std::vector<Corner> m_corners;
+    
+public:
+  /// \endcond
+
+  /*!
+    Constructs a structured point set based on the input points and the
+    associated shape detection object.
+
+    \note Both property maps can be omitted if the default constructors of these property maps can be safely used.
+
+  */
+  Point_set_with_structure (Input_iterator begin, ///< iterator over the first input point.
+                            Input_iterator end, ///< past-the-end iterator over the input points.
+                            Point_map point_map, ///< property map: value_type of InputIterator -> Point_3. 
+                            Normal_map normal_map, ///< property map: value_type of InputIterator -> Vector_3. 
+                            const Shape_detection_3::Efficient_RANSAC<Traits>&
+                            shape_detection, ///< shape detection object
+                            double epsilon, ///< size parameter
+                            double attraction_factor = 3.) ///< attraction factor
+    : m_traits (shape_detection.traits()),
+      m_point_map(point_map), m_normal_map (normal_map)
+  {
+    for (Input_iterator it = begin; it != end; ++ it)
+      {
+        m_points.push_back (get(m_point_map, *it));
+        m_normals.push_back (get(m_normal_map, *it));
+      }
+      
+    m_indices = std::vector<std::size_t> (m_points.size (), (std::numeric_limits<std::size_t>::max)());
+    m_status = std::vector<Point_status> (m_points.size (), POINT);
+
+    BOOST_FOREACH (boost::shared_ptr<Shape> shape, shape_detection.shapes())
+      {
+        boost::shared_ptr<Plane_shape> pshape
+          = boost::dynamic_pointer_cast<Plane_shape>(shape);
+        
+        // Ignore all shapes other than plane
+        if (pshape == boost::shared_ptr<Plane_shape>())
+          continue;
+        m_planes.push_back (pshape);
+
+        for (std::size_t i = 0; i < pshape->indices_of_assigned_points().size (); ++ i)
+          {
+            m_indices[pshape->indices_of_assigned_points()[i]] = m_planes.size () - 1;
+            m_status[pshape->indices_of_assigned_points()[i]] = PLANE;
+          }
+      }
+
+    run (epsilon, attraction_factor);
+    clean ();
+  }
+
+  /// \cond SKIP_IN_MANUAL
+  Point_set_with_structure (Input_iterator begin, ///< iterator over the first input point.
+                            Input_iterator end, ///< past-the-end iterator over the input points.
+                            const Shape_detection_3::Efficient_RANSAC<Traits>&
+                            shape_detection, ///< shape detection object
+                            double epsilon, ///< size parameter
+                            double attraction_factor = 3.) ///< attraction factor
+    : Point_set_with_structure (begin, end, Point_map(), Normal_map(), shape_detection, epsilon, attraction_factor)
+  {
+  }
+  /// \endcond
+
+  /// \cond SKIP_IN_MANUAL    
+  virtual ~Point_set_with_structure ()
```

Comment:
That's a reflex I have when writing a new class, just in case someone wants to inherit it later. But I guess it's probably not that useful if the attributes are all `private` and not `protected`…